### PR TITLE
ci: add publish workflow for releasing on CI automatically

### LINKED ISSUE #177: Extension at Open VSX Registry
### Subject of the feature

Open VSX is a vendor-neutral open-source alternative to the Visual Studio Marketplace. Code - OSS and VSCodium both use this registry.

### Problem

This extension is not available at Open VSX now.

### Expected behavior

It would be great if this extension could be added to Open VSX.

### LINKED ISSUE #188: Republish this as unifiedjs
### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Problem

This extension was first published as [silvenon.mdx](https://marketplace.visualstudio.com/items?itemName=silvenon.mdx). Now it’s published as [JounQin.vscode-mdx](https://marketplace.visualstudio.com/items?itemName=JounQin.vscode-mdx).

I think it makes sense to republish it one more time as [unifiedjs](https://marketplace.visualstudio.com/publishers/unifiedjs) to signal this is endorced by unified.

In combination with https://github.com/remarkjs/vscode-remark/issues/67 this also means it gets the _Verified Publisher_ tag.

### Solution

1. Republish this as unifiedjs
2. Unpublish the old extensions

### Alternatives

Keep it as-is :shrug:

### PR DESCRIPTION
**Pull request by JounQin**
close #177
close #188

@asbjornu Can you please help to review the publishing workflow?

@wooorm An `OPEN_VSX_TOKEN` and a `VSCE_TOKEN` secrets are required.

### CODE REVIEW

**Review by silvenon — COMMENTED** [review]
Other stuff looks 👌

File: .github/workflows/publish.yml
Comment (JounQin) [reply .github/workflows/publish.yml:14]: Based on https://github.com/remarkjs/vscode-remark/blob/main/.github/workflows/publish.yml#L13

It could generate CHANGELOG later.

Comment (JounQin) [reply .github/workflows/publish.yml:14]: Personally I'd like to use https://github.com/changesets/changesets for releasing, I may refactor this part tomorrow.

Comment (silvenon) [reply .github/workflows/publish.yml:14]: Sure! `fetch-depth: 0` to me looked like `git clone --depth=0`, but in fact it's the opposite 😆 So I just wanted to check if you knew that.

**Review by wooorm — APPROVED** [review]
I don’t see anything weird, but I have no experience with publishing to vscode.

I have completely forgotten how to get those tokens. Any ideas? Must I add them before this can be merged?

**Comment by JounQin** [comment]
> I have completely forgotten how to get those tokens. Any ideas? Must I add them before this can be merged?

Right. For vsce, it should be added at `https://dev.azure.com/{account}/_usersSettings/tokens`, for `open-vsx`, I'm also not so familiar too. Maybe @asbjornu can offer some help.

**Comment by JounQin** [comment]
@wooorm 

And also, I'd like to enable https://github.com/changesets/changesets for auto releasing with better CHANGELOG, just sent an installation request, can you help to accept?

**Comment by JounQin** [comment]
https://github.com/EclipseFdn/open-vsx.org/issues/423

It seems you've claimed already for open-vsx.

**Comment by wooorm** [comment]
I approved the two applications you requested on the two repos you maintain here

> can you give me the open-vsx namespace permission?

I added you as a contributor on open-vcx. Which permissions do you mean?

> can you add me into the azure [organization](https://dev.azure.com/unifiedjs) with correct permission?

You already are a contributor in that organization. What else do you need.

> can you enable my permission to manage the Actions secrets?

As discussed before, I don’t think that it's safe for all members to access all information in the unified organizations that has many different contributors. We already have a releasers team in the MDX organization that has access to that, and we have the core team that has access to that. I think we should have a small number of people that have access to vulnerable information.

There is already a `VSCE_TOKEN` secret here from March 2021 btw.

**Comment by wooorm** [comment]
> For vsce, it should be added at https://dev.azure.com/{account}/_usersSettings/tokens

I believe that all the docs on vscode extensions are out of date. These URLs and guides don’t work for me.

**Comment by JounQin** [comment]
@wooorm Great to know! Then the last thing left should be `OPEN_VSX_TOKEN`?

**Comment by JounQin** [comment]
> I added you as a contributor on open-vcx. Which permissions do you mean?

image

Besides, I see no namespace available to me on open-vsx.org still.

**Comment by wooorm** [comment]
> There is already a VSCE_TOKEN secret here from March 2021 btw.

Did you add this a year ago? Perhaps its old and not for `unifiedjs`?

> Then the last thing left should be OPEN_VSX_TOKEN 

I added a `OPEN_VSX_TOKEN` token.

> Besides, I see no namespace available to me on open-vsx.org still.

Screen Shot 2022-07-10 at 19 31 24

**Comment by JounQin** [comment]
> Did you add this a year ago? Perhaps its old and not for unifiedjs?

I have no permission to add secret.

The token will be expired in one year, that's a question.

**Comment by JounQin** [comment]
@wooorm 

https://dev.azure.com/{account}/_usersSettings/tokens

This link is not outdated and works to me, maybe you have a different username for Microsoft account?

**Comment by JounQin** [comment]
https://github.com/remarkjs/vscode-remark/pull/40#issuecomment-803913133

Previous context that may help. @wooorm

**Comment by wooorm** [comment]
> maybe you have a different username for Microsoft account?

I indeed have different accounts. None of those accounts can access that page.
My account connected to GitHub can access pages outside of dev.azure.com.

> The token will be expired in one year, that's a question.

If I added it a year ago, I don’t think so. I am guessing it will work.

**Comment by JounQin** [comment]
https://github.com/mdx-js/vscode-mdx/runs/7279615594?check_suite_focus=true#step:6:20

@wooorm Please help to see, maybe a new token is required for a new extension.

**Comment by JounQin** [comment]
It's already available at https://open-vsx.org/extension/unifiedjs/vscode-mdx, but not https://marketplace.visualstudio.com/items?itemName=unifiedjs.vscode-mdx

**Comment by wooorm** [comment]
https://github.com/microsoft/vsmarketplace/issues/354#issuecomment-1078827787

**Comment by JounQin** [comment]
> Perhaps we can find a workaround where we can move your extension to the unifiedjs publisher “scope”?

I hope we can do that also, but found no solution yet.

**Comment by asbjornu** [comment]
What's the current status here and how may I be of assistance? :)

File: .github/workflows/publish.yml
Comment (remcohaszing) [inline .github/workflows/publish.yml:29]: This action has failed. Can I help to fix this somehow?

File: .github/workflows/publish.yml
Comment (remcohaszing) [inline .github/workflows/publish.yml:23]: Are there any benefits to use this action instead of the official `vsce` CLI?

Also: nice that this extension is also published to the [Open VSX registry](https://open-vsx.org/extension/unifiedjs/vscode-mdx).

Comment (JounQin) [reply .github/workflows/publish.yml:23]: It keeps usage consistency between vsce and open-vsx, I supposed?

**Comment by asbjornu** [comment]
Have a look at the discussion in https://github.com/remarkjs/vscode-remark/issues/28. I don't remember exactly how we did it, but I think by reading the comments you should figure out what we did to move `vscode-remark` from @mrmlnc to `unifiedjs`.

**Comment by asbjornu** [comment]
Ok, then I have no idea, sorry. 🤷🏼

**Comment by remcohaszing** [comment]
Perhaps we can try uploading the extension to the VSCode markerplace manually.

Alternatively we could rename the extension from `vscode-mdx` to `mdx`. There are some discussion for this as well in remarkjs/vscode-remark#66. The main reason to not do it was because it would be a churn, but given the current situation it might actually be convenient for this extension.

**Comment by wooorm** [comment]
I’m quite like releasing it as `mdx`!

**Comment by JounQin** [comment]
https://marketplace.visualstudio.com/items?itemName=unifiedjs.vscode-mdx v1.0.1 is finally been released!

**Pull request merged**
